tests/qos/files/cisco/qos_param_generator.py

```diff
@@ -418,3 +423,43 @@ def __define_pg_drop(self):
                       "pkts_num_margin": margin,
                       "iterations": 100}
             self.write_params("pg_drop", params)
+
+    def __define_xon_hysteresis(self):
+        if self.is_t2:
+            return
```

Comment:
We need support for T2 also going forward.

Reply:
@abdosi I have analyzed 3 linecards of T2, they are not applicable for the same xon hysteresis cases.

T2 skip reasons:

Vanguard Linecard (Q200/Gibraltar, mac-sec, long-link) 

On vanguard, lossless will be evicted to HBM once queue occupancy > 3K buffers (1.125MB), that leaves counter-A usage is mostly for lossy traffic. 

It requires hundreds of lossy flows to do the counter-A region transition 88MB -> 86MB. Because, after sms usage > 81MB, lossy tail drop starts at 19200B. 

 

Lancer Linecard (Q200/Gibraltar, non-mac-sec, short-link) 

Same rx cgm_profile as Churchill-mono, Lancer is not able to support the same xon hysteresis testcases since not enough ports. There are 6 ports usable in testing per asic, considering port-channel config. 

 

Guantlet Linecard (Q100/Pacific) 

On pacific, lossy will be evicted to HBM once queue occupancy > 50 buffers (19200 bytes). The SQG0 usage and counter-A usage are all for lossless traffic. 

For pacific, SQG usage is per slice, current sonic-mgmt script infra doesn’t support choosing ports from a specific slice. There are several port-channels containing two ports from two different slices, which makes choosing port from single slice even more difficult. Thus, skip Gauntlet.

Reply:
@abdosi can you please update

Reply:
FYI @abdosi  Please review